content/Software and Downloads/Upload/Error-avrdude-when-uploading.md

```diff
@@ -63,17 +69,111 @@ avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
+1. Make sure you have [selected the right board and port](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE) correct board under _Tools > Board_ and  _Tools > Port_.
+
+   * For instructions, see [Select board and port in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE)
+
+   * If your board is missing, see [If your board does not appear in the port menu](https://support.arduino.cc/hc/en-us/articles/4412955149586-If-your-board-does-not-appear-in-the-port-menu)
+
+2. If using the classic Nano, try selecting a different processor in _Tools > Processor_.
+
+3. For some boards, a _loopback test_ can be performed, which will rule out problems with the serial chip. It can be done with the following boards:
+
+   * Arduino Uno (classic)
+   * Arduino Uno Rev3
+   * Arduino Uno Rev3 SMD
+   * Arduino Mega (classic)
+   * Arduino Mega2560 Rev3
+   * Arduino Mega ADK Rev3
+   * Arduino Nano (classic)
+
+   If the loopback test passes, or if you cant do a loopback test, reprogramming the bootloader may resolve the issue. [ArduinoISP](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP) can be used for boards with AtMega chips. You can also search for [articles in the Help Center](https://support.arduino.cc/hc/en-us/search?utf8=%E2%9C%93&query=bootloader).
+
+---
+
+<a id="stk500v2_ReceiveMessage_getsync"></a>
+
+## avrdude: stk500v2_ReceiveMessage(): timeout<br>avrdude: stk500v2_getsync(): timeout communicating with programmer
+
+Make sure you have [selected the right board and port](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE) correct board under _Tools > Board_ and  _Tools > Port_.
+
+---
+
+<a id="avrdude-ser_open-cant-open-device-no-such-file-or-directory"></a>
+
+## avrdude: ser_open(): can't open device: No such file or directory
+
+```
+avrdude: ser_open(): can't open device "/dev/cu.usbmodem14101": No such file or directory
+Problem uploading to board. See http://www.arduino.cc/en/Guide/Troubleshooting#upload for suggestions.
+```
+
+This error can happen if the board on the selected port has been disconnected from the computer, or reassigned to a different port.
+
+1. Make sure the correct port is selected in _Tools > Port_.
+
+   * For step-by-step instructions, see [Select board and port in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE).
+   * If you can't find the port, see [If your board does not appear in the port menu](https://support.arduino.cc/hc/en-us/articles/4412955149586-If-your-board-does-not-appear-in-the-port-menu).
+
+2. If using the classic Nano, try selecting a different processor in _Tools > Processor_.
+
+> On Windows, you can [manually set a port for the board](https://support.arduino.cc/hc/en-us/articles/360016420140#set-COM-port) to stop it from being reassigned.
```

Comment:
This link will work when the article is updated with df205d1c6ba6e38528890988930f2c565a90f523